resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/FileStoreUtil.java

```diff
@@ -135,9 +135,15 @@ private static int compareStringOrNull(String string1, String string2) {
 	 * @return 0 if the fileStores are equal, 1 if fileStore1 is bigger than fileStore2, -1 otherwise
 	 */
 	public static int compareFileStore(IFileStore fileStore1, IFileStore fileStore2) {
-		int compare = compareStringOrNull(fileStore1.getFileSystem().getScheme(), fileStore2.getFileSystem().getScheme());
-		if (compare != 0)
-			return compare;
+		try {
+			String scheme1 = fileStore1.getFileSystem().getScheme();
+			String scheme2 = fileStore2.getFileSystem().getScheme();
+			int compare = compareStringOrNull(scheme1, scheme2);
+			if (compare != 0)
+				return compare;
+		} catch (NullPointerException e) {
+			throw new RuntimeException("fileStore1" + fileStore1.getClass() + ':' + fileStore1 + " fileStore2:" + fileStore2.getClass() + ':' + fileStore2, e); //$NON-NLS-1$//$NON-NLS-2$
```

Comment:
I acknowledge that the stack trace showed that `getFileSystem` returned null, so it is probably ok. But this line here will create another NPE if fileStore1 or fileStore2 are null.

Note that the message could use some spaces, too.

What do you think about retaining the original exception type:
```
NullPointerException enhanced = new NullPointerException(msg);
enhanced.initCause(e);
throw enhanced;
```

Reply:
Something like this:

```java
String msg = String.format("%s (fileStore1: %s; fileStore2: %s)", e.getMessage, fileStore1== null? null : fileStore1.getClass().getName(), fileStore2== null? null : fileStore2.getClass().getName());
NullPointerException enhanced = new NullPointerException(msg);
enhanced.initCause(e);
throw enhanced;
```

Reply:
does not matter, it snot checked exception anyway. the goal is only to get the wrong implementation class to then fix the underlying error